### PR TITLE
Use RWLock for containers

### PR DESCRIPTION
**Pull request by mrunalp**
Addresses part of #1824.
@runcom PTAL

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: oci/container.go [hunk 1]
```diff
@@ -241,8 +240,8 @@ func (c *Container) Metadata() *pb.ContainerMetadata {
 
 // State returns the state of the running container
 func (c *Container) State() *ContainerState {
```
Comment (runcom) [inline oci/container.go]: is it just this operation?

**Comment by rhatdan** [comment]
@mrunalp How do I make myself worthy of the `/lgtm`

**Comment by mrunalp** [comment]
@rhatdan I think it is looking at the OWNERS file. Can you make a PR to add yourself to it?

**Comment by rhatdan** [comment]
@mrunalp Nope, I am in OWNERS.  Must be something about my name in kubernetes-sig or Openshift?

**Pull request merged**